client/lib/domains/constants.js

```diff
@@ -17,7 +17,40 @@ const registrar = {
 	MAINTENANCE: 'Registrar TLD Maintenance'
 };
 
+const tlds = {
```

Comment:
If we call `/products` we should simply populate the redux store instead of `tlds` which is local to this module.
I think you're right about the maintenance cost, this was a hacky solution to get this out quickly but since this PR took some delay anyway for other reasons I think we may be able to do this in this PR.